frontend/src/components/explorer/MapViewer.vue

```diff
@@ -11,24 +11,31 @@
              class="column is-one-fifth-widescreen is-one-quarter-desktop
                     is-one-quarter-tablet has-background-lightgray om-2
                     fixed-height-desktop scrollable">
-          <div
-            class="buttons has-addons is-centered" @click="$store.dispatch('maps/toggleShowing2D')"
-            tooltip="haha">
-            <button :class="showing2D ? 'is-selected is-primary has-text-weight-bold' : 'is-light'"
-                    class="button" title="Switch to 3D">2D</button>
-            <button :class="!showing2D ? 'is-selected is-primary has-text-weight-bold' : 'is-light'"
-                    class="button" title="Switch to 2D">3D</button>
+          <div id="dimensionToggle" class="buttons has-addons is-centered padding-mobile"
+               :title="`Switch to ${dimensionalState(!showing2D) }`"
+               @click="$store.dispatch('maps/toggleShowing2D')" tooltip="haha">
+            <button v-for="dim in [true, false]" :key="dim"
+                    class="button"
+                    :class="dim === showing2D ? 'is-selected is-primary has-text-weight-bold' : 'is-light'">
+              <span v-if="dim === showing2D" class="icon">
+                <i class="fa fa-check-square-o"></i>
+              </span>
+              <span v-if="dim !== showing2D">Switch to&nbsp;</span>
```

Comment:
This sometimes causes buttons to be in two rows but I think it's negligible for now.

Reply:
I think it's the `min-width` causing that, but removing that makes the text go outside the borders of the buttons; indeed, negligible for now.

Reply:
Yes I meant specifically the change in text. It might fit without the “Switch to” addition.

Reply:
I tried, but it had a `width` that had to be overridden, so it made no difference. I kept that size for `min-width` instead.